api/ChattO/API/Controllers/BillingController.cs

```diff
@@ -1,27 +1,43 @@
 ﻿using API.DTOs.Requests.Billing;
 using API.DTOs.Responses.Billing;
 using API.Helpers;
+using Application.Abstractions;
 using Application.Helpers;
+using Application.Payment.Commands;
+using Application.Payment.Queries;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace API.Controllers;
 
 public class BillingController : BaseController
 {
-    //Create billing by organization
-    [Authorize(Roles = RolesConstants.SuperAdmin)]
+    private readonly IUserService _userService;
+
+    public BillingController(IUserService userService)
+    {
+        _userService = userService;
+    }
+
+    [Authorize(Roles = RolesConstants.Admin)]
     [HttpPost]
     [ProducesResponseType<bool>(StatusCodes.Status200OK)]
     [ProducesResponseType<string>(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> Create([FromBody] CreateBillingRequest request)
+    public async Task<IActionResult> Create([FromQuery] Guid billingInfoId)
     {
-        return Ok();
+        var user = await _userService.GetCurrentUser();
+        var request = new CreateBillingRequest() { OrganizationId = user.Data.OrganizationId, BillingInfoId = billingInfoId };
```

Comment:
Why do we need this additional entity? We can just pass needed properties  to command

Reply:
done